83010 - Appointment Detail null States

ryanshaw opened the pull request:
## Summary

This updates appointment detail pages to better handle situations where the appointments application does not recieve various appointment, location and facility data.


## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83010

## Testing done

- Manual Testing
- Updated unit tests

## Local Testing
To help with local testing and validation, new mocks have been added to [`src/applications/vaos/services/mocks/index.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/eef9af7cf8684e6afe15cb77c049501786ebf15e/src/applications/vaos/services/mocks/index.js#L35)

Here's the [Figma file](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2159-51275&t=pxGzhSukzuC02chF-1)

## Screenshots



|              | With Data | Missing Data |
| ------- | ------- | ----- |
| In-Person |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/b8353ba1-f688-43b3-b785-e929b0999da6)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/4f0af836-ae78-4357-b327-cd678031122d)|
|Video|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/138f313c-61b5-4e38-ad13-8b6b0cb01dcd)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/5d3dd0e9-f752-4cc2-9185-1f6c3d137a55)|
|VA Video|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/7f13fc10-996b-4147-bb80-c6937b676086)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/fb9de9d8-4f46-4e29-87ee-093cae62b87c)|
|ATLAS|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/07da1e44-980c-4d2a-9e2a-c8fbe3cc7e41)|![image](https://github.com/user-attachments/assets/938e4ba8-31a4-4e15-8095-40d2021e038b)|
|Community Care|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/9d05e256-7ff5-43bc-9d50-608061ea38ff)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/be01f95f-a2c9-49b8-a4a4-66b40204b710)|
|Phone|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/c0c36bfa-2788-4fe4-9914-dd09884c2d1a)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/8b495f96-9aac-400d-8dc5-b74c9b8f1c57)|
|Claim Exam|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/c2be7a93-aef9-4736-b2ea-58510c82fdf0)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587812/e95eeb8d-eae2-4eea-a447-72a9d27b79e9)|







## What areas of the site does it impact?

This impacts the following appointment detail pages:
- In-person appointment
- Video
- Video at VA
- ATLAS
- Phone
- Community Care
- Claim Exam

## Acceptance criteria

Background: the `va_online_scheduling_appointment_details_redesign` toggle is on

- [ ] Given when the services return a null value for type of care for all appointment types
When the user is on the Appointment Details page
Then the label `What` will not be displayed 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

- [ ] Given when the services return a null value for the facility (but receive the facility ID) for all appointment types
When the user is on the Appointment Details page
Then the frontend will display the facility name
And a link to facility locator 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

- [ ] Given when the services return a null value for the facility (no facility ID) for all appointment types
When the user is on the Appointment Details page
Then the frontend will display the text `Facility details not available`
And a link to the facility locator 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

### Quality Assurance & Testing

- [ ] I |updated and added unit tests and integration tests for each feature 
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

ryanshaw committed:
Adds mocks for testing null states locally

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Turns appointment page redesign feature flag on

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixes typo

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates null states fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Removes unused null state fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates null states on new detail page layouts

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Include all mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixed fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Update fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Changes clinic phone header to phone and updates text

The clinic phone display will be addressed in this issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/80375

This temporarily changes the header to the correct header for the facility phone for the time being.

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Changes clinic phone header to phone and updates text

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates unit tests for not showing type of care when null is returned

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fix typo

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates proptypes

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates unit tests to handle Phone vs Clinic Phone

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updated null state fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

ryanshaw committed:
Updated null state mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Return locationId

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Removes unneeded setup data

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds test coverage for Who headings in VA video appointments

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates VA video layout  unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates VA video layout  unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates phone layout unit test

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates null state mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates VideoLayoutVA unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixes phone heading in claim exam layout unit test

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Reorganizes va video unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds tests to in person layout for missing data

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds ATLAS video unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixes claim exam layout unit test

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates fixtures

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates VideoLayout unit test

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Added phonelayout unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds null state request mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds CCLayout unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates request mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Revert "Updates request mocks"

This reverts commit 60233164adece5ca04e8018d5c6e4ae423fc35be.

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Toggle mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds past appointment

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates CCLayout unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Reorganizes missing data unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates VideoLayout and tests as per spec

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adjusts mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Turns off null state mocks by default

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates video layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates ATLAS layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

ryanshaw committed:
Merge branch 'main' into ryanshaw/83010-null-state-appointment-details

jenniemc reviewed (commented):
For video atlas appt, when there is missing address location it currently displays "Where to attend" heading but there's no text underneath. Should there be message that say something like "address is not available"? Everything else is looking good.
![Screenshot 2024-07-09 at 4 01 50 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/54327023/2a402629-1fe7-46f8-b594-e03595ea2bae)

JunTaoLuo commented:
A few thoughts/questions after going through the figma and the screenshots above:

## General

In several of appointment types, there's a section for "Who" that should display the provider name when the data is available. I don't see those sections in the "With Data" screenshots. Is that intentional?

## VA Video appointments

For VA Video appointments, I think we should check with the designers what's the expectation for the "Need to make changes?" section. Specifically, like the Video appointments, the specification at the top of the figma file says:
<img width="353" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/2030323/ff26fee0-3cff-4ef3-8228-b31007702ab1">
Which indicates that "Facility not available" and "Clinic not available" should be displayed but, unlike the Video appointments, the example at the bottom of the figma file does not have this section. 
<img width="396" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/2030323/321c5d7d-2131-444a-8b21-adee2359d693">

## ATLAS

In the specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

## Phone appointments

For phone appointments, the figma files says:
<img width="347" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/2030323/6b4de1e7-28d9-484f-afc1-bacfff5afd85">
Is that something we need to add?
In the specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

## Community Care

This is a question for the designers. For every other appointment type that has provider information, it is displayed under the "Who" heading but Community Care appointments display it under the "Provider" heading. Is this intentional?

## Claim Exam

The screenshot still has a "What" section can you take a screenshot of when that's missing? Also, the specifications calls for using the main VA number when the facility phone is unavailable.

I'll start digging into the code and business logic next.

jenniemc commented:
Good catch @JunTaoLuo 

> In several of appointment types, there's a section for "Who" that should display the provider name when the data is available. I don't see those sections in the "With Data" screenshots. Is that intentional?

Some appointment types such as in-person and claims exam, did not bake the "who" into the layout. We need to create a new ticket to ensure those appointments display the Who.

> In the ATLAS specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

The set of mock data are pull directly from staging. If the `serviceType` (what) for ATLAS is missing it is because it was not returned. We need to check the vets-api If ATLAS appt has the serviceType property returned. If so we can simply add serviceType into our mock ATLAS data.

> The Claim exam screenshot still has a "What" section can you take a screenshot of when that's missing? 

For Claim exam appointments, the "What" will always default to Claim exam and will ignore the `serviceType` property. Therefore the "What" will always be present 

> Also, the specifications calls for using the main VA number when the facility phone is unavailable.

Phone numbers will be fixed in the ticket, [#80375](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80375)

JunTaoLuo commented:
> For Claim exam appointments, the "What" will always default to Claim exam and will ignore the serviceType property. Therefore the "What" will always be present

Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?

> The set of mock data are pull directly from staging. If the serviceType (what) for ATLAS is missing it is because it was not returned. We need to check the vets-api If ATLAS appt has the serviceType property returned. If so we can simply add serviceType into our mock ATLAS data.

Ah I see, let me see if I can dig into this a bit in vets-api. Might take me a minute to figure it out though.

jenniemc commented:
> Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?

When I saw the [figma mockup](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=3270-47171&t=h5AfSLX3paE54SUd-0) I thought it was repetitive as well to have Claim exam repeated in the title and in the "What". This would be a great question to ask @Cieramaddox or @outerpress if it is intentional.

ryanshaw committed:
Adds clinic to ATLAS video layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Removes extra br from atlas video layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds facility not found on ATLAS appointment layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw commented:
> For video atlas appt, when there is missing address location it currently displays "Where to attend" heading but there's no text underneath. Should there be message that say something like "address is not available"? Everything else is looking good. ![Screenshot 2024-07-09 at 4 01 50 PM](https://private-user-images.githubusercontent.com/54327023/347176830-2a402629-1fe7-46f8-b594-e03595ea2bae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NjExNDUsIm5iZiI6MTcyMDY2MDg0NSwicGF0aCI6Ii81NDMyNzAyMy8zNDcxNzY4MzAtMmE0MDI2MjktMWZlNy00NmY4LWI1OTQtZTAzNTk1ZWEyYmFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzExVDAxMjA0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3N2YxZjU1OTY2ZTAxZTg1OWZhMGMzYTUzZjAzOTAxOWY1YzAwYzNhY2VjNjFkYjlhNmQ4YjMzYmU2NzA3MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SqU_eqFGgvd8isihAVkPR2o4JvTi_ZAAcHkwGFaOS1c)

@jenniemc Just pushed a fix for this and updated screenshots

ryanshaw commented:
> > Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?
> 
> When I saw the [figma mockup](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=3270-47171&t=h5AfSLX3paE54SUd-0) I thought it was repetitive as well to have Claim exam repeated in the title and in the "What". This would be a great question to ask @Cieramaddox or @outerpress if it is intentional.

@jenniemc @JunTaoLuo I talked to @Cieramaddox about this and it is intentional to have both for consistency purposes. We can revisit it at a later date if needed.

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

jenniemc commented:
ATLAS video appointments are sometimes held at a non VA medical facility such as an American Legion Post or at a Veteran Foreign War Post. The "find facility link" takes the user to VA Service type and not necessarily to ATLAS locator. A better link would be https://telehealth.va.gov/atlas. Please confirm with the Ciera which link to use and what language to use for missing ATLAS location address.

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

JunTaoLuo commented on src/applications/vaos/components/layout/InPersonLayout.jsx:
```diff
@@ -64,24 +68,43 @@ export default function InPersonLayout({ data: appointment }) {
+              <br />
+            </>
+          )}
+        {/* When the services return a null value for the facility (but receive the facility ID) */}
```
In the spec it says:

<img width="337" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/2030323/d0be499f-bfbe-4fce-af67-7da02065c0ee">

Is this the facility name something we can retrieve using just the ID?

ryanshaw replied:
Current behavior (on old design) doesn't display the name and I cannot find anywhere that would exist if we only have the id. My assumption is that we wouldn't be getting that information back from the facilities endpoint but would be able to construct the link with the id to allow users to get more information about that facility.

JunTaoLuo replied:
That's fair enough, but might want to check with @Cieramaddox to make sure.

JunTaoLuo commented on src/applications/vaos/components/layout/InPersonLayout.jsx:
```diff
@@ -98,7 +121,7 @@ export default function InPersonLayout({ data: appointment }) {
         <br />
         {facilityPhone && (
-          <FacilityPhone heading="Clinic phone:" contact={facilityPhone} />
+          <FacilityPhone heading="Phone:" contact={facilityPhone} />
```
Hmm, the spec seems to prefer clinic phone and falls back to facility phone if that's missing. Do we need to update the logic here?

<img width="344" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/2030323/595d1c5b-5cb5-46ce-957a-e2ecad2ef346">

ryanshaw replied:
@JunTaoLuo All the phone number issues are in another issue that @jenniemc is working on (https://github.com/department-of-veterans-affairs/va.gov-team/issues/80375)

JunTaoLuo replied:
Ah I see it's tracked in the same issue. Thanks!

JunTaoLuo reviewed (commented):
Looks great so far. Just had a few more questions in the code. Looking at the tests now and that should be the last bit of review I need to do.

JunTaoLuo commented on src/applications/vaos/components/layout/CCLayout.jsx:
```diff
@@ -57,12 +57,14 @@ export default function CCLayout({ data: appointment }) {
         </When>
-        <What>{typeOfCareName || 'Type of care not noted'}</What>
+        <What>{typeOfCareName}</What>
         <Section heading="Provider">
```
Hey @Cieramaddox I see that in the figma that all other appointments uses the "Who" heading to display the provider information with CC being the only type that uses the "Provider" heading. Just wanted to double check if this is intentional.

JunTaoLuo commented on src/applications/vaos/components/tests/ClaimExamLayout.unit.spec.js:
```diff
@@ -33,6 +33,66 @@ describe('VAOS Component: ClaimExamLayout', () => {
     },
   };
 
+  describe('When appointment information is missing', () => {
```
Can we add another case for when facility ID is present but facility info is missing? This goes for all other appointment types that has a different "View facility information"/"Find facility information" distinction.

ryanshaw replied:
@JunTaoLuo I've added tests cases for this for `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx`

JunTaoLuo commented on src/applications/vaos/components/tests/PhoneLayout.unit.spec.js:
```diff
@@ -33,6 +33,78 @@ describe('VAOS Component: PhoneLayout', () => {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Also need to add assertions for the text for missing facility/clinic info as well as the reason/other details texts.

ryanshaw replied:
@JunTaoLuo Added assertions for the phone appointments

JunTaoLuo commented on src/applications/vaos/components/tests/VideoLayoutAtlas.unit.spec.js:
```diff
@@ -33,6 +33,79 @@ describe('VAOS Component: VideoLayoutAtlas', () => {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Add assertions for missing facility and clinic info.

ryanshaw replied:
@JunTaoLuo I've added these assertions

JunTaoLuo commented on src/applications/vaos/components/tests/VideoLayoutVA.unit.spec.js:
```diff
@@ -33,6 +33,68 @@ describe('VAOS Component: VideoLayoutVA', () => {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Need to add assertions for missing facility and clinic info. Also consider adding another case where facility id is present but missing facility info.

ryanshaw replied:
@JunTaoLuo I've added tests cases for this (the facility id/missing facility scenario) for `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx`

JunTaoLuo reviewed (commented):
Looks good. Some additional testing details to add.

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

ryanshaw committed:
Updates copy on VARequestLayout, CCRequestLayout and CancelPageLayoutRequest

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds more null state mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks for local testing

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixes typo

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds missing facility assertion to atlas layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates test assertions for video layout va

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates assertions for facility info on in person layout unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates unit test assertions on unit tests for c&p appointment layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Merge branch 'main' of github.com:department-of-veterans-affairs/vets-website into ryanshaw/83010-null-state-appointment-details

ryanshaw committed:
Removes bad mock data

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates mocks

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds not found response to facilities mock

When we have a facility id (i.e. locationId in the response) but do not have any facility information.

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Revert "Adds not found response to facilities mock"

This reverts commit e7c0783eff6f6420c4d747d7cc008402b9e55f4f.

ryanshaw committed:
Updates unit tests for inpersonlayout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates claim exam location/facilities and unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds more mock test data for null state

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates video va unit test and layout

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Fixes placement of phone and clinic information when no facility data is available

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw commented on src/applications/vaos/components/layout/ClaimExamLayout.jsx:
```diff
@@ -69,25 +72,44 @@ export default function ClaimExamLayout({ data: appointment }) {
-            <br />
-          </>
-        )}
+        {/* When the services return a null value for the facility (no facility ID) for all appointment types */}
```
The facility logic in `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx` should probably be pulled out and placed in a component on its own. Could simplify unit tests as well as future code changes. Didn't want to introduce that refactor this late in the review process so will open another issue for that.

ryanshaw committed:
Updates phone layout unit tests to check for clinic

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Removes phone not available text

This is taken care of in another issue/PR

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Added assertions to phone layout unit test for reason and other details

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Adds assertions for clinic and facility fields on atlas video layout unit test

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Updates bad setup for location data on in person and claim exam layout unit tests

Signed-off-by: Ryan Shaw <ryan.shaw@adhocteam.us>

ryanshaw committed:
Merge branch 'main' into ryanshaw/83010-null-state-appointment-details

JunTaoLuo reviewed (approved):
LGTM! There just two things to check with design but otherwise I think this is ready 😄. Hey @vbahinwillit @jenniemc can you take a look at this as well?

jenniemc reviewed (approved):
LGTM! Let it merge!!
